src/main/java/de/tum/in/www1/artemis/repository/LectureRepository.java

```diff
@@ -44,10 +44,20 @@ public interface LectureRepository extends JpaRepository<Lecture, Long> {
             FROM Lecture lecture
             LEFT JOIN FETCH lecture.posts
             LEFT JOIN FETCH lecture.lectureUnits lu
+            LEFT JOIN FETCH lu.completedUsers cu
             LEFT JOIN FETCH lu.learningGoals
             WHERE lecture.id = :#{#lectureId}
             """)
-    Optional<Lecture> findByIdWithPostsAndLectureUnitsAndLearningGoals(@Param("lectureId") Long lectureId);
+    Optional<Lecture> findByIdWithPostsAndLectureUnitsAndLearningGoalsAndCompletions(@Param("lectureId") Long lectureId);
+
+    @Query("""
+            SELECT lecture
+            FROM Lecture lecture
+            LEFT JOIN FETCH lecture.lectureUnits lu
+            LEFT JOIN FETCH lu.learningGoals
+            WHERE lecture.id = :#{#lectureId}
```

Comment:
```suggestion
            WHERE lecture.id = :lectureId
```
Same here :thinking: